### DISCUSSION TITLE
Static files dir hosting

### DISCUSSION THREAD

**Discussion by bmarcj** [tokio-rs/axum#1822] [Q&A] [answered]
I'm struggling to understand axum's static file hosting. This works:
```
  let app = Router::new()
        .route("/static", get_service(ServeDir::new("./static/"))
            .handle_error(|error: io::Error| async move {
                (
                    StatusCode::INTERNAL_SERVER_ERROR,
                    format!("Unhandled internal error: {}", error),
                )
            }));
```
But this (straight from the examples) fails with ``` expected enum `Infallible`, found struct `std::io::Error` ```
```
 let app = Router::new()
        .nest_service("/static", ServeDir::new("./static/"));
```
What am I doing wrong? And how does this differ from the examples in repo?     
Thanks!

**Comment by davidpdrsn** [comment, marked as answer]
You're using an old version of tower-http most likely. Make sure you're using the same version as the examples (0.4)

**Reply by bmarcj** [reply]
Gah, it's that simple. Thanks.
